contrib/gin-gonic/gin/appsec.go

```diff
@@ -27,7 +27,8 @@ func useAppSec(c *gin.Context, span tracer.Span) func() {
 		}
 	}
 	args := httpsec.MakeHandlerOperationArgs(req, params)
-	op := httpsec.StartOperation(args, nil)
+	ctx, op := httpsec.StartOperation(c.Request.Context(), args)
+	*req = *req.WithContext(ctx)
```

Comment:
I think we avoid a struct copy by doing:
```suggestion
	c.Request = req.WithContext(ctx)
```